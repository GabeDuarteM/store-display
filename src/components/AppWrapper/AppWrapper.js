import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import { Provider } from 'react-redux'

export const styles = {
  [`@global ${
    process.env.NODE_ENV === 'styleguide' ? '*[data-preview]' : 'body'
  }`]: {
    fontFamily: 'Montserrat, sans-serif',
    backgroundColor: '#f5f5f5',
  },
}

const AppWrapper = ({ children, store }) => {
  if (store) {
    return <Provider store={store}>{children}</Provider>
  }
  return children
}

export const propTypes = {
  /** The children that the component will render */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  /** The redux's store */
  store: PropTypes.object,
}

export const defaultProps = {
  store: null,
}

AppWrapper.propTypes = propTypes
AppWrapper.defaultProps = defaultProps

export default injectSheet(styles)(AppWrapper)
