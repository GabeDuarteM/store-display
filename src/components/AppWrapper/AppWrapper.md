A simple wrapper that injects everything needed for the app to run.

This is shared with Styleguidist, wrapping every component in a AppWrapper, applying the same styles and configs whenever possible

```js
<AppWrapper>
  thing to be <span>wrapped</span>
</AppWrapper>
```
