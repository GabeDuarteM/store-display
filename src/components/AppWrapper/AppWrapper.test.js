import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AppWrapper, { styles, propTypes } from './AppWrapper'

const createWrapper = (
  func,
  children = (
    <React.Fragment>
      thing to be <span>wrapped</span>
    </React.Fragment>
  ),
) => func(<AppWrapper>{children}</AppWrapper>)

const createWrapperShallow = children => createWrapper(shallow, children)

const createWrapperMount = children => createWrapper(mount, children)

describe('COMPONENT: <AppWrapper />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it('should correctly render whatever is passed as children inside the wrapper', () => {
    const children = <div>é bolacha, não biscoito (brazilian joke)</div>
    const wrapper = createWrapperShallow(children)
    expect(wrapper.dive().containsAllMatchingElements([children])).toBeTruthy()
  })

  it('should render global styles under the key "@global body" when process.env.NODE_ENV is different than "styleguide"', () => {
    // eslint-disable-next-line global-require
    const { styles: stylesDev } = require('./AppWrapper')

    expect(process.env.NODE_ENV).not.toBe('styleguide')
    expect(stylesDev['@global body']).toBeDefined()
  })

  it('should render global styles under the key "@global *[data-preview]" when process.env.NODE_ENV is "styleguide"', () => {
    const nodeEnvBackup = process.env.NODE_ENV

    process.env.NODE_ENV = 'styleguide'
    jest.resetModules()

    // eslint-disable-next-line global-require
    const { styles: stylesStyleguide } = require('./AppWrapper')

    expect(process.env.NODE_ENV).toBe('styleguide')
    expect(stylesStyleguide['@global *[data-preview]']).toBeDefined()

    process.env.NODE_ENV = nodeEnvBackup
    jest.resetModules()
    expect(process.env.NODE_ENV).toBe(nodeEnvBackup)
  })
})
