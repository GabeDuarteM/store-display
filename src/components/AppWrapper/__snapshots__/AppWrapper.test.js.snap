// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <AppWrapper /> should match component's propTypes snapshot 1`] = `
Object {
  "children": [Function],
  "store": [Function],
}
`;

exports[`COMPONENT: <AppWrapper /> should match component's snapshot 1`] = `
<React.Fragment>
  thing to be 
  <span>
    wrapped
  </span>
</React.Fragment>
`;

exports[`COMPONENT: <AppWrapper /> should match component's style snapshot 1`] = `
Object {
  "@global body": Object {
    "backgroundColor": "#f5f5f5",
    "fontFamily": "Montserrat, sans-serif",
  },
}
`;
